internal/pkg/runtime/launcher/oci/process_linux.go

```diff
@@ -103,42 +103,45 @@ func (l *Launcher) getProcessCwd() (dir string, err error) {
 // userns from which the OCI runtime is launched.
 //
 //	e.g. host 1001 -> fakeroot userns 0 -> container targetUID
-func (l *Launcher) getReverseUserMaps(targetUID, targetGID uint32) (uidMap, gidMap []specs.LinuxIDMapping, err error) {
-	uid := uint32(os.Getuid())
-	gid := uint32(os.Getgid())
+func getReverseUserMaps(hostUID, targetUID, targetGID uint32) (uidMap, gidMap []specs.LinuxIDMapping, err error) {
 	// Get user's configured subuid & subgid ranges
-	subuidRange, err := fakeroot.GetIDRange(fakeroot.SubUIDFile, uid)
+	subuidRange, err := fakeroot.GetIDRange(fakeroot.SubUIDFile, hostUID)
 	if err != nil {
 		return nil, nil, err
 	}
 	// We must always be able to map at least 0->65535 inside the container, so we cover 'nobody'.
 	if subuidRange.Size < 65536 {
 		return nil, nil, fmt.Errorf("subuid range size (%d) must be at least 65536", subuidRange.Size)
 	}
-	subgidRange, err := fakeroot.GetIDRange(fakeroot.SubGIDFile, uid)
+	subgidRange, err := fakeroot.GetIDRange(fakeroot.SubGIDFile, hostUID)
 	if err != nil {
 		return nil, nil, err
 	}
 	if subgidRange.Size < 65536 {
 		return nil, nil, fmt.Errorf("subuid range size (%d) must be at least 65536", subgidRange.Size)
 	}
 
-	if uid < subuidRange.Size {
+	uidMap, gidMap = reverseMapByRange(targetUID, targetGID, *subuidRange, *subgidRange)
+	return uidMap, gidMap, nil
+}
+
+func reverseMapByRange(targetUID, targetGID uint32, subuidRange, subgidRange specs.LinuxIDMapping) (uidMap, gidMap []specs.LinuxIDMapping) {
```

Comment:
I have to confess, I couldn't completely follow why the (newly created) `reverseMapByRange()` is doing what it does, but I did look to make sure that it's preserving the logic of the code that it was refactored from. 👍